traffic_ops/traffic_ops_golang/riaksvc/riak_services.go

```diff
@@ -241,40 +296,103 @@ func RiakServersToCluster(servers []ServerAddr, authOptions *riak.AuthOptions) (
 	}
 	opts := &riak.ClusterOptions{
 		Nodes:             nodes,
-		ExecutionAttempts: MaxCommandExecutionAttempts,
+		ExecutionAttempts: DefaultMaxCommandExecutionAttempts,
 	}
 	cluster, err := riak.NewCluster(opts)
 	if err != nil {
 		return nil, errors.New("creating riak cluster: " + err.Error())
 	}
+	return cluster, err
+}
+
+func RiakServersToCluster(servers []ServerAddr, authOptions *riak.AuthOptions) (StorageCluster, error) {
+	cluster, err := riakServersToClusterPriv(servers, authOptions)
+	if err != nil {
+		return nil, err
+	}
+	return RiakStorageCluster{Cluster: cluster}, nil
+}
+
+func GetPooledClusterTx(tx *sql.Tx, authOptions *riak.AuthOptions) (StorageCluster, error) {
+
+	clusterMutex.Lock()
+	defer clusterMutex.Unlock()
+
+	var newservers []ServerAddr
+	tryLoad := false
+
+	timenow := time.Now()
+
+	// should we try to reload the cluster?
```

Comment:
I don't think we need the extra complexity here to only recheck the Riak servers once every X seconds or whatever. We should be fine to just check the riak servers unconditionally on every Riak-required request and rebuild the shared cluster if the servers list has changed. The cost of an extra DB call on each request is negligible I think and worth it to have Riak server updates known immediately.